py_experimenter/experimenter.py

```diff
@@ -29,58 +29,101 @@ def __init__(self,
+        :type value: str
         """
-        if not self._config.has_section(section_name):
-            self._config.add_section(section_name)
-        self._config.set(section_name, key, value)
-        PyExperimenter._valid_configuration(self._config, self._credential_path)
+        if not self.config.has_section(section_name):
+            self.config.add_section(section_name)
+        self.config.set(section_name, key, value)
+        PyExperimenter._is_valid_configuration(self.config, self.credential_path)
 
     def get_config_value(self, section_name: str, key: str) -> str:
-        return self._config.get(section_name, key)
+        """
+        Returns the value of the experiment configuration for the given `key`. If the `key` is not contained within the section, 
+        an exception is raised.
+
+        :param section_name: The name of the section of the experiment configuration of which a value should be returned.
+        :type section_name: str
+        :param key: The name of the key within the given section of the experiment configuration of which a value should be returned
+        :type key: str
+        :return: The value of the `key` field of the experiment configuration within the section having `section_name`.
+        :rtype: str
+        :raises NoOptionError: If the `key` is not contained within the section.
+        """
+        return self.config.get(section_name, key)
 
     def has_option(self, section_name: str, key: str) -> bool:
-        return self._config.has_option(section_name, key)
+        """
+        Checks whether the experiment configuration contains the given `key` within the section called `section_name`.
+
+        :param section_name: The name of the section of the experiment configuration of which the `key` should be checked.
+        :type section_name: str
+        :param key: The name of the key to check within the given section.
+        :type key: str
+        :return: True if the given `key` is contained in the experiment configuration within the section called `section_name`.
+        :rtype: bool
+        """
+        return self.config.has_option(section_name, key)
 
     @ staticmethod
-    def _valid_configuration(_config: configparser, credential_path=None) -> bool:
+    def _is_valid_configuration(_config: configparser, credential_path: str = None) -> bool:
         """
-        This method checks if the configuration is valid.
+        Checks whether the given experiment configuration is valid, i.e., it contains all necessary fields, 
+        and in case of a mysql database provider, that the database credentials are available.
+
+        :param _config: The experiment configuration.
+        :type _config: configparser
+        :param credential_path: The path to the database configuration file, i.e., the file defining the host, user and password. Defaults to None.
```

Comment:
Not sure, if "credential_path" is really a good name. How about "database_credential_file_path"?